content/blog/2019/root-transition/index.md

```diff
@@ -0,0 +1,142 @@
+After the transition, the root certificate will be renewed to have 10 year lifetime.
+As a best practice for security, we strongly recommend you to do annual root key/cert rotation.
+We will send out instructions for root key/cert rotation as a follow-up.
+
+To evaluate the lifetime remaining for your root certificate, please refer to Step 0 in the following procedure.
+
+We provide the following procedure for you to do the root transition.
+Note that the Envoy instances will be hot restarted to reload the new root certificates, which may impact the long-lived connections.
+For details about the impacts and how Envoy hot restart works, please refer to
+[here](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/hot_restart#arch-overview-hot-restart) and
+[here](https://blog.envoyproxy.io/envoy-hot-restart-1d16b14555b5).
+
+## Scenarios
+
+If you are not currently using the mTLS feature in Istio and will not use it in the future,
+you are not affected and should only upgrade to 1.1.8 or later versions.
+
+If you are not currently using the mTLS feature in Istio and may use it in the future,
+you are recommended to follow the procedure listed below to upgrade.
+
+If you are currently using the mTLS feature in Istio with self-signed certificates,
+please follow the procedure and check whether you will be affected.
+
+## Procedure
+
+### Step 0. Check when your root certificate will expire
+
+Download this [script](TODO) on a machine that has kubectl access to the cluster.
+
+{{< text bash>}}
+$ ./root-transition.sh check
+...
+===YOU HAVE 30 DAYS BEFORE THE ROOT CERT EXPIRED!=====
+{{< /text >}}
+
+Please run the following steps to upgrade immediately to avoid system outages, before your root certificate expires.
+
+### Step 1. Do the root certificate transition
+
+Run the following command to do the root certificate transition.
+
+{{< text bash>}}
+$ ./root_transition.sh transition
+create new ca cert, with trust domain as  cluster.abc
+Wed Jun  5 19:11:15 PDT 2019 delete old ca secret
+secret "istio-ca-secret" deleted
+Wed Jun  5 19:11:15 PDT 2019 create new ca secret
+secret/istio-ca-secret created
+pod "istio-citadel-8574b88bcd-j7v2d" deleted
+Wed Jun  5 19:11:18 PDT 2019 restarted Citadel, checking status
+NAME                             READY     STATUS    RESTARTS   AGE
+istio-citadel-8574b88bcd-l2g74   1/1       Running   0          3s
+New root certificate:
+Certificate:
+    Data:
+        ...
+        Validity
+            Not Before: Jun  6 03:24:43 2019 GMT
+            Not After : Jun  3 03:24:43 2029 GMT
+        Subject: O = " cluster.abc"
+        ...
+{{< /text >}}
+
+After the transition, the Envoy sidecars may be hot-restarted. This may have some impact on your traffic.
+Please refer to 
+[here](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/hot_restart#arch-overview-hot-restart) and
+[here](https://blog.envoyproxy.io/envoy-hot-restart-1d16b14555b5) for more details.
```

Comment:
here/read this(link) blog post for more details.

Reply:
Done.